src/vivliostyle/selectors.js

```diff
@@ -181,4 +183,231 @@ goog.scope(function() {
+        var doc = parentNodeContext.viewNode.ownerDocument;
+        var fragment = doc.createDocumentFragment();
+        var viewRoot = doc.createElement("div");
+        var pseudoColumn = new PseudoColumn(column, viewRoot, parentNodeContext);
+        var initialPageBreakType = pseudoColumn.getColumn().pageBreakType;
+        pseudoColumn.getColumn().pageBreakType = null;
+        return pseudoColumn.layout(this.createNodePositionForPseudoElement(), true).thenAsync(function() {
+            this.styler.contentProcessed["after-if-continues"] = false;
+            pseudoColumn.getColumn().pageBreakType = initialPageBreakType;
+            var pseudoElement = /** @type {!Element} */ (viewRoot.firstChild);
+            adapt.base.setCSSProperty(pseudoElement, "display", "block");
+            return adapt.task.newResult(pseudoElement);
+        }.bind(this));
+    };
+
+    /**
+     * @private
+     * @return {!adapt.vtree.ChunkPosition}
+     */
+    AfterIfContinues.prototype.createNodePositionForPseudoElement = function() {
+        var sourceNode = adapt.vgen.pseudoelementDoc.createElementNS(adapt.base.NS.XHTML, "div");
+        adapt.vgen.setPseudoName(sourceNode, "after-if-continues");
+        var shadowContext = this.createShadowContext(sourceNode);
+        var step = {
+            node: sourceNode,
+            shadowType: shadowContext.type,
+            shadowContext: shadowContext,
+            nodeShadow: null,
+            shadowSibling: null
+        };
+        var nodePosition = {steps:[step], offsetInNode:0, after:false, preprocessedTextContent:null};
+        return new adapt.vtree.ChunkPosition(nodePosition);
+    };
+
+    /**
+     * @private
+     * @param {!Element} root
+     * @return {!adapt.vtree.ShadowContext}
+     */
+    AfterIfContinues.prototype.createShadowContext = function(root) {
+        return new adapt.vtree.ShadowContext(this.sourceNode, root, null, null,
+            null, adapt.vtree.ShadowType.ROOTED, this.styler);
+    };
+
+    /**
+     * @constructor
+     * @param {!adapt.vtree.NodeContext} nodeContext
+     * @param {!vivliostyle.selectors.AfterIfContinues} afterIfContinues
+     * @param {number} pseudoElementHeight
+     * @implements {adapt.layout.FragmentLayoutConstraint}
+     */
+    vivliostyle.selectors.AfterIfContinuesLayoutConstraint = function(
+        nodeContext, afterIfContinues, pseudoElementHeight) {
+        this.nodeContext = nodeContext;
+        this.afterIfContinues = afterIfContinues;
+        this.pseudoElementHeight = pseudoElementHeight;
+    };
+    /** @const */ var AfterIfContinuesLayoutConstraint = vivliostyle.selectors.AfterIfContinuesLayoutConstraint;
+
+    /** @override */
+    AfterIfContinuesLayoutConstraint.prototype.allowLayout = function(nodeContext, overflownNodeContext, column) {
+        if ((overflownNodeContext && !nodeContext) || (nodeContext && nodeContext.overflow)) {
+            return false;
```

Comment:
Is is necessary to return `false` here?
Since `AfterIfContinuesLayoutConstraint` has no "next candidate" (`nextCandidate()` does nothing), it seems to me that stating "layout of this `NodeContext` is not allowed" by returning `false` here is useless.
In the test case (`after-if-continues.html`), when this condition is true, [`RepetitiveElementsOwnerLayoutConstraint#allowLayout()` returns `false`](https://github.com/vivliostyle/vivliostyle.js/blob/b256acb5ee936279be012f4dd87269ded8ec85cf/src/vivliostyle/repetitiveelements.js#L580) too.
Therefore removing the condition here (always returning `true`) does not change layout result.